Sources/Danger/GitLabDSL.swift

```diff
@@ -0,0 +1,222 @@
+import Foundation
+
+public struct GitLab: Decodable {
+    public enum CodingKeys: String, CodingKey {
+        case mergeRequest = "mr"
+        case metadata
+    }
+
+    public let mergeRequest: GitLabMergeRequest
+    public let metadata: GitLabMetadata
+}
+
+public struct GitLabMetadata: Decodable, Equatable {
+    public let pullRequestID: String
+    public let repoSlug: String
+}
+
+public struct GitLabMergeRequest: Decodable, Equatable {
+    public enum State: String, Decodable {
+        case closed
+        case locked
+        case merged
+        case open
+    }
+
+    public struct Milestone: Decodable, Equatable {
+        public enum CodingKeys: String, CodingKey {
+            case createdAt = "created_at"
+            case description
+            case dueDate = "due_date"
+            case id
+            case iid
+            case projectId = "project_id"
+            case startDate = "start_date"
+            case state
+            case title
+            case updatedAt = "updated_at"
+            case webUrl = "web_url"
+        }
+
+        public enum State: String, Decodable {
+            case closed
+        }
+
+        public let createdAt: Date
+        public let description: String
+        public let dueDate: Date
+        public let id: Int
+        public let iid: Int
+        public let projectId: Int
+        public let startDate: Date
+        public let state: State
+        public let title: String
+        public let updatedAt: Date
+        public let webUrl: String
+    }
+
+    public struct TimeStats: Decodable, Equatable {
+        public enum CodingKeys: String, CodingKey {
```

Comment:
<!--
  0 failure: 
  1 warning:  Types should be n...
  
  
  DangerID: danger-id-Danger;
  File: Sources/Danger/GitLabDSL.swift;
  Line: 59;
-->

- :warning: Types should be nested at most 1 level deep (`nesting`)